docs/rest_api.rst

```diff
@@ -0,0 +1,951 @@
+Let's dive into a simple example using the quickhowto.
+The quickhowto example as a Contact's Model and a Group Model, so each Contact belongs to a Group.
+
+First let's define a CRUD REST Api for our Group model resource::
+
+    from flask_appbuilder.models.sqla.interface import SQLAInterface
+    from flask_appbuilder.api import ModelRestApi
+    from . import appbuilder
+
+
+    class GroupModelRestApi(ModelRestApi):
+        resource_name = 'group'
+        datamodel = SQLAInterface(ContactGroup)
+
+    appbuilder.add_view_no_menu(MyFirstApi)
+
+Behind the scenes FAB uses marshmallow-sqlalchemy to infer the Model to a Marshmallow Schema,
+that can be safely serialized and deserialized. Let's recall our Model definition for ``ContactGroup``::
+
+    class ContactGroup(Model):
+        id = Column(Integer, primary_key=True)
+        name = Column(String(50), unique=True, nullable=False)
+
+        def __repr__(self):
+            return self.name
+
+
+All endpoints are protected so we need to request a JWT and use it on our REST resource,
+like shown before we need to make a PUT request to the login API endpoint::
+
+    # Login to obtain a token
+    $ curl -XPOST http://localhost:8080/api/v1/security/login -d \
+      '{"username": "admin", "password": "password", "provider": "db"}' \
+      -H "Content-Type: application/json"
+    {
+      "access_token": "<SOME TOKEN>"
+    }
+    # It's nice to use the Token as an env var
+    $ export TOKEN="<SOME TOKEN>"
+
+First let's create a Group::
+
+    $ curl -XPOST http://localhost:8080/api/v1/group/ -d \
+     '{"name": "Friends"}' \
+     -H "Content-Type: application/json" \
+     -H "Authorization: Bearer $TOKEN"
+    {
+      "id": 1,
+      "result": {
+        "name": "Friends"
+      }
+    }
+
+We got back a response with the model id and result with the inserted data.
+Now let's query our newly created Group::
+
+    $ curl http://localhost:8080/api/v1/group/1 \
+     -H "Content-Type: application/json" \
+     -H "Authorization: Bearer $TOKEN"
+
+    {
+      "description_columns": {},
+      "include_columns": [
+        "name"
+      ],
+      "label_columns": {
+        "name": "Name"
+      },
+      "id": "1",
+      "result": {
+        "name": "Friends"
+      }
+    }
+
+As you can see, the API returns the model data, and extra meta data so you can properly render
+a page with labels, descriptions and defined column order. This way it should be possible
+to develop a React component (for example) that renders any model just by switching between HTTP endpoints.
+It's also possible to just ask for certain meta data keys, we will talk about this later.
+
+Next let's change our newly created model (HTTP PUT)::
+
+    $ curl -XPUT http://localhost:8080/api/v1/group/1 -d \
+     '{"name": "Friends Changed"}' \
+     -H "Content-Type: application/json" \
+     -H "Authorization: Bearer $TOKEN"
+    {
+      "result": {
+        "name": "Friends Changed"
```

Comment:
Quick question here: if the model has many fields and I pass only a subset of those fields in my post, only those get updated? I remember I fixed the old update endpoint in earlier FAB to have this behavior.

Reply:
Yes, it can work like HTTP PATCH

```
curl http://localhost:8080/api/v1/contact/1 \         
-H "Content-Type: application/json" \
-H "Authorization: Bearer $TOKEN"
{
  "description_columns": {}, 
  "id": "1", 
  "include_columns": [
    "contact_group", 
    "gender", 
    "name", 
    "address", 
    "birthday", 
    "personal_phone", 
    "personal_celphone"
  ], 
  "label_columns": {
    "address": "Address", 
    "birthday": "Birthday", 
    "contact_group": "Contact Group", 
    "gender": "Gender", 
    "name": "Name", 
    "personal_celphone": "Personal Celphone", 
    "personal_phone": "Personal Phone"
  }, 
  "result": {
    "address": "", 
    "birthday": "2019-03-22", 
    "contact_group": 1, 
    "gender": 1, 
    "name": "contact1", 
    "personal_celphone": "11111", 
    "personal_phone": "99999"
  }
}

curl -XPUT http://localhost:8080/api/v1/contact/1 -d \
'{"name": "contact1 changed"}' \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $TOKEN"
{
  "result": {
    "address": "", 
    "birthday": "2019-03-22", 
    "contact_group": 1, 
    "gender": 1, 
    "name": "contact1 changed", 
    "personal_celphone": "11111", 
    "personal_phone": "99999"
  }
}

```